src/ctype/CTypeMetadata.spec.ts

```diff
@@ -0,0 +1,51 @@
+import CType from './CType'
+import ICType, { ICtypeMetadata } from '../types/CType'
+import CTypeMetadata from './CTypeMetadata'
+import * as CTypeUtils from './CTypeUtils'
+import { CTypeWrapperMetadata } from './CTypeSchema'
+
+jest.mock('../blockchainApiConnection/BlockchainApiConnection')
+
+describe('CType', () => {
+  const ctype = new CType({
+    schema: {
+      $id: 'http://example.com/ctype-1',
+      $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+      properties: {
+        'first-property': { type: 'integer' },
+        'second-property': { type: 'string' },
+      },
+      type: 'object',
+    },
+  } as ICType)
+
+  const ctypeMetadata = {
+    metadata: {
+      $id: 'http://example.com/metadata-1',
+      $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+      title: { type: 'string' },
+      description: { type: 'string' },
+      properties: {
+        'first-property': { type: 'integer' },
+        'second-property': { type: 'string' },
+      },
+      type: 'object',
+    },
+    ctypeHash: 'string',
+  } as ICtypeMetadata
+
+  const metadata = new CTypeMetadata(ctype, ctypeMetadata)
+
```

Comment:
I am not so thrilled about putting the whole ctype within the CTypeMetadata, creating the new instances. 

This was my work-around, maybe there is a better way of doing it. 

Or will we need it to build the ctyeMetadata?
It seems messy and contains too much unnecessary information.

Reply:
I don't understand. Where are you putting the whole ctype somewhere?

Reply:
I wanted to ask if we should just use the CTypeHash or a whole CType into the metadata?

So, inside the Metadata, we only use the CTypeHash. What do you think?